torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -43,6 +43,11 @@ static std::vector<std::string> TORCH_NCCL_ASYNC_ERROR_HANDLING = {
     "TORCH_NCCL_ASYNC_ERROR_HANDLING",
     "NCCL_ASYNC_ERROR_HANDLING"};
 
+// Environment Variable to control whether Desync Debug is enabled.
+// This variable must be set together with TORCH_NCCL_ASYNC_ERROR_HANDLING.
+static std::vector<std::string> TORCH_NCCL_DUMP_ON_TIMEOUT = {
+    "TORCH_NCCL_DESYNC_DEBUG"};
```

Comment:
Wait, are we reusing desync flag for the dump?

Reply:
good catch, thats broken!

Reply:
https://github.com/pytorch/pytorch/pull/114699